pkg/reconciler/revision/rate_limiter.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package revision
+
+import (
+	"math"
+	"sync"
+	"time"
+
+	"k8s.io/client-go/util/workqueue"
+)
+
+// itemExponentialFailureRateLimiter does a simple baseDelay*2^<num-failures> limit
+// dealing with max failures and expiration are up to the caller
+//
+// When num-failures is 0 it will not wait
+//
+// Copyright 2016 The Kubernetes Authors.
+// From: https://github.com/kubernetes/client-go/blob/master/util/workqueue/default_rate_limiters.go
+//
+// TODO - drop this file when we pick up the following upstream change
+// https://github.com/kubernetes/kubernetes/pull/108343
```

Comment:
Is this file effectively a fork of upstream at this PR?

Reply:
Yup